packages/core/src/index.ts

```diff
@@ -212,6 +212,16 @@ export type CreateEnv<
   >
 >;
 
+const getDefaults = <Schema extends AnyZodObject>(schema: Schema) => {
+  return Object.fromEntries(
+    Object.entries(schema.shape).map(([key, value]) => {
+      if (value instanceof ZodDefault)
```

Comment:
oh i actualy like this, but don't think it'll be supported with this change: https://github.com/t3-oss/t3-env/pull/299 ://

Reply:
unfortunately no - you could possibly add a `getDefault` option that receives a given schema and returns a default if it can get one?

Reply:
an alternative would be something like

```ts
createEnv({
  server: {
    FOO: v.optional(v.string)
  },
  runtimeEnv: {
    FOO: process.env.FOO ?? "DEFAUTL"
  }
})
```

Reply:
I'd bet for @EskiMojo14 solution, as the goal should be to avoid declaring the same thing twice ;)

Reply:
@EskiMojo14 really liked your solution, but just wondering if we should introduce new property that's required, I think that it could be optional and by default, we'll return all the default values, wdyt?

Reply:
@Bartek532 [done](https://github.com/EskiMojo14/t3-env/commit/770ce3da61df02c42adba490954dbeb32c40f616) - had the same thought

Reply:
Looks good! @EskiMojo14 could you open the PR for this? We can then review together with @juliusmarminge ;)

Reply:
#313 isn't merged, so if I opened the PR right now it would be on my fork not this one.